third_party/amd/lib/TritonAMDGPUTransforms/AccelerateAMDMatmul.cpp

```diff
@@ -319,21 +342,46 @@ class BlockedToMFMA : public mlir::RewritePattern {
     auto oldAcc = dotOp.getOperand(2);
     auto newAcc = convertAndCastTensor(rewriter, oldAcc, mfmaEnc, mfmaAccType);
 
-    // kWidth is a number of consecutive elements per one instruction per one
-    // thread
-    auto kWidth = -1;
-    // in mfma 32x32 case argument matrix groups elements in 2 groups
-    // in mfma 16x16 case argument matrix groups elements in 4 groups
+    // Here is a brief explanation of kWidth, kBase, and kDim
+    // 1. kWidth: the number of elements each thread loads from shared memory in
+    //    preparation for mfma instructions. In theory each thread can issue one
+    //    or more load instructions to load a total of kWidth elements, since
+    //    those elements are not required to be in contiguous addresses in
+    //    shared memory. But in practice, we make sure the kWidth elements can
+    //    be loaded from shared memory by a single ds_read instruction by
+    //    setting vecSize of the sharedLayout to be kWidth.
+    // 2. kDim: the k dimension size of the mfma instruction. E.g. instruction
+    //    mfma_32x32x16 has kDim = 16, meaning this mfma instruction can compute
+    //    a matmul of operands with shape 32x16 and 16x32.
+    // 3. kBase: the number of elements each thread holds for a single mfma
+    //    instruction.
+    // 4. relation between kBase and kDim:
+    //    4.1 For mfma_32, kBase = kDim / 2
+    //    4.2 For mfma_16, kBase = kDim / 4
+    //    4.3 For mfma_4, it depends on how mfma_4 is used. We'll extend to
+    //        mfma_4 later.
+    // 5. relation between kWidth and kBase: For now it supports two cases
+    //    5.1 kWidth = kBase, i.e. kPack = 1. In this case, each load from
+    //        shared memory results in one mfma instruction.
+    //    5.2 kWidth = 2 * kBase, i.e. kPack = 2. In this case, each load from
+    //        shared memory results in two mfma instructions, since one mfma
+    //        can only consume kBase elements from each thread.
+    //    Note that we cannot have larger kPack since kPack = 2 means
+    //    ds_read_b128, which is the largest vector size for shared memory load.
```

Comment:
@zahimoud I added some doc here. Please let me know if you are still confused by these k* variables or you want me to move this to somewhere else.